lib/attrsets.nix

```diff
@@ -55,10 +55,13 @@ rec {
        => { a = { b = 3; }; }
   */
   setAttrByPath = attrPath: value:
-    if attrPath == [] then value
-    else listToAttrs
-      [ { name = head attrPath; value = setAttrByPath (tail attrPath) value; } ];
-
+    let
+      len = length attrPath;
+      foldr = n:
```

Comment:
It isn't really `foldr` anymore. How about `atDepth`?

Reply:
fixed, thanks